### PR TITLE
Improvements: Allow Designer to run locally with remote debugging & normalize NUSSKNACKER_SCALA_VERSION env param name

### PR DESCRIPTION
**Pull request by lciolecki**
- passing -agentlib:jdwp=transport=dt_socket,server=y,suspend=n,address at run.sh
- replace SCALA_VERSION by NUSSKNACKER_SCALA_VERSION

## Checklist before merge
- [ ] Related issue ID is placed at the beginning of PR title in \[brackets\] (can be GH issue or Nu Jira issue)
- [x] Code is cleaned from temporary changes and commented out lines
- [ ] Parts of the code that are not easy to understand are documented in the code
- [x] Changes are covered by automated tests
- [ ] Showcase in dev-application.conf added to demonstrate the feature
- [ ] Documentation added or updated
- [x] Added entry in _Changelog.md_ describing the change from the perspective of a public distribution user
- [x] Added _MigrationGuide.md_ entry in the appropriate subcategory if introducing a breaking change
- [ ] Verify that PR will be squashed during merge

### CODE REVIEW

File: docs/Changelog.md [hunk 1]
```diff
@@ -1,5 +1,8 @@
 # Changelog
 
+1.15.0 (Not released yet)
+* [#5724](https://github.com/TouK/nussknacker/pull/5724) Improvements: Allow Designer to run locally with remote debugging & normalize SCALA_VERSION env param name
```
Comment (arkadius) [inline docs/Changelog.md:4]: I think that we shouldn't expose this in changelog. It is interesting for developers instead of users. Instead, we should mention about that in contributing guide

File: docs/MigrationGuide.md [hunk 2]
```diff
@@ -26,6 +26,9 @@ To see the biggest differences please consult the [changelog](Changelog.md).
 ### Other changes
 
 * [#5574](https://github.com/TouK/nussknacker/pull/5574) Removed the support for the plugable expression languages: `ExpressionConfig.languages` removed
+* [#5724](https://github.com/TouK/nussknacker/pull/5724) Improvements: Run Designer locally
+  * Introduce `JAVA_DEBUG_PORT` to run the Designer locally with remote debugging capability
+  * Removed `NUSSKNACKER_SCALA_VERSION` from running Designer locally, please use `SCALA_VERSION` instead
```
Comment (arkadius) [inline docs/MigrationGuide.md:31]: Why did you change it from `NUSSKNACKER_SCALA_VERSION` to `SCALA_VERSION`? Thanks to nussknacker prefix, it was possible to set up this on the system level (in `.profile` file). I used it quite often when for some reasons I had to switch to 2.12 for a longer period.

Comment (lciolecki) [reply docs/MigrationGuide.md:31]: Hmm, ok it has sense.. But right now we use SCALA_VERSION on run scripts like ./runServer / CI scripts and NUSSKNACKER_SACLA_VERSION on sbt.. IMO, we should normalize it and use one convention everywhere. We can stay with NUSSKNACKER_SCALA_VERSION, WDYT?

**Review by arkadius — APPROVED** [review]
